docs/ref/models/fields.txt

```diff
@@ -1031,6 +1031,83 @@ are converted to lowercase.
+
+A field for storing JSON encoded data. In Python the data is represented in
+its Python native format: dictionaries, lists, strings, numbers, booleans
+and ``None``.
+
+.. attribute:: JSONField.encoder
+
+    An optional :py:class:`json.JSONEncoder` subclass to serialize data types
+    not supported by the standard JSON serializer (``datetime``, ``uuid``,
+    etc.). For example, you can use the
+    :class:`~django.core.serializers.json.DjangoJSONEncoder` class.
+
+    If ``None`` is specified, it defaults to ``json.JSONEncoder``.
+
+.. attribute:: JSONField.decoder
+
+    An optional :py:class:`json.JSONDecoder` subclass to deserialize the value
+    retrieved from the database. The value will be in the format chosen by the
+    custom encoder (most often a string). Your deserialization may need to
+    account for the fact that you can't be certain of the input type. For
+    example, you run the risk of returning a ``datetime`` that was actually a
+    string that just happened to be in the same format chosen for
+    ``datetime``\s.
+
+    If ``None`` is specified, it defaults to ``json.JSONDecoder``.
+
+Specification for ``JSONEncoder`` and ``JSONDecoder`` subclasses can be found
+in the Python documentation.
+
+If you give the field a :attr:`~django.db.models.Field.default`, ensure
+it's an immutable object, such as a ``str``, or a callable object that returns
+a fresh mutable object each time, such as ``dict`` or a function. Providing a
+mutable default object like ``default={}`` or ``default=[]`` shares the one
+object between all model instances. There is a system check for this.
+
+To query ``JSONField`` in the database, see
+:ref:`Querying JSONField <querying-jsonfield>`.
+
+.. note::
+
+    Oracle Database supports storing JSON data using SQL data types
+    ``VARCHAR2``, ``CLOB``, and ``BLOB``. ``JSONField`` uses ``CLOB``.
+
+.. note::
+
+    PostgreSQL has two native JSON based data types: ``json`` and ``jsonb``.
+    The main difference between them is how they are stored and how they can be
+    queried. PostgreSQL's ``json`` field is stored as the original string
+    representation of the JSON and must be decoded on the fly when queried
+    based on keys. The ``jsonb`` field is stored based on the actual structure
+    of the JSON which allows indexing. The trade-off is a small additional cost
+    on writing to the ``jsonb`` field. ``JSONField`` uses ``jsonb``.
+
+.. admonition:: MariaDB users
+
+    Requires MariaDB 10.2.7 or later.
+
+.. admonition:: MySQL users
+
+    Requires MySQL 5.7.8 or later.
+
+.. admonition:: Oracle users
+
+    Oracle Database does not support storing JSON scalar values. Only JSON
+    objects and arrays (represented in Python using :py:class:`dict` and
+    :py:class:`list`) are supported.
+
+.. admonition:: SQLite users
+
+    Requires SQLite 3.9.0 or later with the
+    `JSON1 <https://www.sqlite.org/json1.html>`_ extension enabled. See
+    :doc:`/howto/sqlite-json1` for more details.
```

Comment:
This link would need updating to `refs/database#sqlite-notes` (or whatever the rST syntax is) after my suggested move :)